Library/Homebrew/formula.rb

```diff
@@ -922,6 +897,32 @@ def plist_startup
     self.class.plist_startup
   end
 
+  # This method can be used to provide a plist.
+  # @see https://www.unix.com/man-page/all/5/plist/ Apple's plist(5) man page
+  # <pre>plist do
+  #  {
+  #    Label: "#{plist_name}.override",
+  #    ProgramArguments: ["#{opt_bin}/example", "--do-this"],
+  #    RunAtLoad: true,
+  #    KeepAlive: true,
+  #    StandardErrorPath: "/dev/null",
+  #    StandardOutPath: "/dev/null",
+  #  }
+  # end</pre>
+  def plist
```

Comment:
It should still return a string when no argument is given.

Reply:
@reitermarkus The [previous](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/formula.rb#L895-L897) version of this method was literally just
~~~ rb
def plist
  nil
end
~~~
so it only returned `nil` and didn't take any arguments… only there to be overriden if specified in a formula (which do currently return a raw xml string - the point of this PR being to avoid having to do that eventually) in which case `formula.plist` when called from `formula_installer.rb` is not `nil` and so passes the `return` and gets written. I don't see how this breaks that because `def plist` in a formula still just overrides this method and works as before… Have you tested it? Both [multiple] `plist do` blocks and `def plist` (overriden mehtods) in a test formula work as expected for me. Am I missing something here?